src/github.com/getlantern/flashlight/flashlight.go

```diff
@@ -253,9 +253,9 @@ func runClientProxy(cfg *config.Config) {
 	watchDirectAddrs()
 
 	go func() {
-		addExitFunc(pacOff)
 		err := client.ListenAndServe(func() {
 			pacOn()
+			addExitFunc(pacOff)
```

Comment:
Even when `ListenAndServe()` failed the `pacOff()` function was called anyway. It's better to move it here after `pacOn()`.